app/scripts/models/nodes/Node.js

```diff
@@ -377,122 +378,14 @@ define(['backbone', 'FLOOD', 'staticHelpers'], function (Backbone, FLOOD, static
     },
 
     updateNodeGeometry: function(param) {
-        var graphicData = param.geometryData.graphicPrimitivesData;
 
-        graphicData.pointVertices = staticHelpers.getFloatArray(graphicData.pointVertices);
-        graphicData.lineStripVertices = staticHelpers.getFloatArray(graphicData.lineStripVertices);
-        graphicData.lineStripCounts = staticHelpers.getIntArray(graphicData.lineStripCounts);
-        graphicData.triangleVertices = staticHelpers.getFloatArray(graphicData.triangleVertices);
-        graphicData.triangleNormals = staticHelpers.getFloatArray(graphicData.triangleNormals);
-
-        var geometries = []; // prettyLastValue
-        graphicData.numberOfCoordinates = 3;
-
-        this.addPoints(graphicData, geometries);
-        this.addTriangles(graphicData, geometries);
-        this.addCurves(graphicData, geometries);
-
-        this.set('prettyLastValue', geometries);
-        },
-
-        clearGeometry: function() {
-            this.set('prettyLastValue', {});
-    },
-
-    addPoints: function (graphicData, geometries) {
-        var length, pos;
-        // if we have single points
-        if (graphicData.pointVertices && graphicData.pointVertices.length) {
-            length = graphicData.pointVertices.length / graphicData.numberOfCoordinates
-            var points = {vertices: []}, onePoint;
-            for(var i = 0; i < length; i++){
-                // add [x, y, z]
-                pos = i * graphicData.numberOfCoordinates;
-                onePoint = [
-                  graphicData.pointVertices[pos], 
-                  graphicData.pointVertices[pos + 1],
-                  graphicData.pointVertices[pos + 2]
-                ];
-                points.vertices.push(onePoint);
-            }
-
-            geometries.push(points);
-        }
+        this.prettyLastValue = param;
```

Comment:
It occurs when we change an attribute and old geometry is copied to _previousAttributes field. So we always have 2 sets of geometry for any node and it uses a lot of memory.